src/CodeGen/ODataT4CodeGenerator.cs

```diff
@@ -3850,7 +3848,23 @@ internal override void WriteGeneratedEdmModel(string escapedEdmxString)
                 global::System.Xml.XmlReader reader = CreateXmlReader(Edmx);
                 try
                 {
-                    return global::Microsoft.OData.Edm.Csdl.CsdlReader.Parse(reader);
+                    global::System.Collections.Generic.IEnumerable<global::Microsoft.OData.Edm.Validation.EdmError> errors;
+                    global::Microsoft.OData.Edm.IEdmModel edmModel;
+                    
+                    if (!global::Microsoft.OData.Edm.Csdl.CsdlReader.TryParse(reader, ");
+                    this.Write(this.context.IgnoreUnexpectedElementsAndAttributes.ToString().ToLower());
+                    this.Write(@", out edmModel, out errors))
+                    {
+                        global::System.Text.StringBuilder errorMessages = new System.Text.StringBuilder();
+                        foreach (var error in errors)
+                        {
+                            errorMessages.Append(error.ErrorMessage);
+                            errorMessages.Append(""; "");
+                        }
+                        throw new global::System.InvalidOperationException(errorMessages.ToString());

```

Comment:
note: I used a LINQ expression to concatenate the error messages at first but one of the generated files didn't seem to like that so I changed it to this foreach loop.

Reply:
Is that change not part of this PR?

---
In reply to: [197884091](https://github.com/OData/odata.net/pull/1150#discussion_r197884091) [](ancestors = 197884091)